tracer.go

```diff
@@ -121,6 +118,74 @@ type StartSpanOptions struct {
 	// identical to those for Span.SetTag(). May be nil.
 	//
 	// If specified, the caller hands off ownership of Tags at
-	// StartSpanWithOptions() invocation time.
+	// StartSpan() invocation time.
 	Tags map[string]interface{}
 }
+
+// CausalReferenceType is an enum type describing different sorts of
+// relationships between spans. If Span A refers to Span B, the
+// CausalReferenceType describes Span B from Span A's perspective. For example,
+// RefBlockedParent means that Span B is Span A's parent, and that it's blocked
+// on Span A's finish.
+type CausalReferenceType int
+
+const (
+	// RefStartsBefore refers to a span which MUST start before the Span
+	// that's starting.
+	RefStartsBefore CausalReferenceType = iota
+
+	// RefBlockedOnFinish refers to a span which CAN NOT finish successfully
+	// until the Span that's starting has finished.
+	RefBlockedOnFinish
+
+	// RefFinishesBefore refers to a span which MUST finish before the Span
+	// that's starting.
+	RefFinishesBefore
+
+	// RefBlockedParent is the union of RefStartedBefore and RefBlockedOnFinish.
+	RefBlockedParent
+
+	// RefRPCClient is the special case of RefBlockedParent for the RPC client
+	// peer of an RPC server span.
+	RefRPCClient
+
+	// TODO: etc etc, per
+	// https://github.com/opentracing/opentracing.github.io/issues/28
+)
+
+// CausalReference pairs a reference type and a referent SpanContext. See the
+// CausalReferenceType documentation.
+type CausalReference struct {
+	RefType CausalReferenceType
+	SpanContext
+}
+
+// StartSpanOption instances (zero or more) may be passed to Tracer.StartSpan.
+type StartSpanOption func(*StartSpanOptions)
```

Comment:
I would like this to be an interface, not a function. And the options below to be structs, not lambdas.